Automattic/jetpack
# Codecov action: remove token from action

#### Changes proposed in this Pull Request:

Follow-up from #17960. 

It seems that a token may not be needed for public repositories:
> If you have a public project on TravisCI, CircleCI, AppVeyor, Azure Pipelines, or GitHub Actions an upload token is not required.
-- https://docs.codecov.io/docs/about-the-codecov-bash-uploader

#### Jetpack product discussion

* Internal reference: p1610030313084100-slack-CBG1CP4EN
 
#### Does this pull request change what data or activity we track or use?

* No

#### Testing instructions:

* Does the codecov action work?
* Check https://codecov.io/gh/Automattic/jetpack/pull/18230 for the results.

#### Proposed changelog entry for your changes:

* N/A
